graphql-server/src/databases/database.resolver.ts

```diff
@@ -91,6 +115,29 @@ export class DatabaseResolver {
     return dbs;
   }
 
+  @Mutation(_returns => [String])
+  async databasesByConnection(
+    @Args() args: AddDatabaseConnectionArgs,
+  ): Promise<string[]> {
+    if (this.conn.getWorkbenchConfig()?.connectionUrl === args.connectionUrl) {
+      return this.databases();
+    }
+    const workbenchConfig = getWorkbenchConfigFromArgs(args);
+    const ds = getDataSource(workbenchConfig);
+    await ds.initialize();
+    const dbs = await ds.query("SHOW DATABASES");
```

Comment:
Does this work for postgres too? I think the data source might have a `databases` method?